src/project-css/app/Http/Controllers/DataViewController.php

```diff
@@ -91,4 +99,120 @@ public function show(Request $request, $curTable, $curId){
+            ->orderBy('score', 'desc')
+            ->offset($page-1 * $perPage)
+            ->limit($perPage);
+
+    $rcrds = $query
+            ->get(['*', DB::raw("MATCH (srchindex) AGAINST ('".$srchStrng."') AS score")]);
+
+    $rcrdsCount = count($rcrds);
+
+    // if fulltext search yeilds no results try a basic like search
+    // if ($rcrdsCount == 0) {
+    //
+    //   $offsetNum = ($page == 1) ? 0 : ($page-1) * $perPage;
+    //
+    //   // strip characters used by the fulltext boolean search
+    //   //$srchStrng = preg_replace('/[+-]/', '', $srchStrng);
+    //
+    //   $query = DB::table($curTable->tblNme)
+    //           ->where('srchindex', 'like', '%' . $srchStrng . '%')
+    //           ->orderBy('id', 'asc')
+    //           ->offset($offsetNum)
+    //           ->limit($perPage);
+    //
+    //   $rcrds = $query
+    //           ->get();
+    //
+    //   $rcrdsCount = count($rcrds);
+    // }
+
+    // if last query returned exactly 30 items
+    // we assume that their are additional pages
+    // so we set $lastPage to $page + 1
+    $lastPage = ($rcrdsCount == $perPage) ? $page + 1 : $page;
+
+
+    return view('user.search')->with('rcrds', $rcrds)
+                              ->with('clmnNmes', $clmnNmes)
+                              ->with('tblNme', $curTable->tblNme)
+                              ->with('tblId', $curTable)
+                              ->with('search', $srchStrng)
+                              ->with('page', $page)
+                              ->with('lastPage', $lastPage)
+                              ->with('morepages', $page < $lastPage);
+  }
+
+  public function view(Request $request, $curTable, $subfolder, $filename){
+    // Get the table entry in meta table "tables"
+    $curTable = Table::find($curTable);
+
+    if(!$curTable->hasAccess){
+      return redirect()->route('home')->withErrors(['Table is disabled']);
+    }
+
+    // retrieve the column names
+    $clmnNmes = DB::getSchemaBuilder()->getColumnListing($curTable->tblNme);
+
+    $path = storage_path('app/' . $curTable->tblNme . '/' . $subfolder . '/' . $filename);
+
+    return Response::make(file_get_contents($path), 200, [
+        'Content-Type' => Storage::getMimeType($curTable->tblNme . '/' . $subfolder . '/' . $filename),
+        'Content-Disposition' => 'inline; filename="'.$filename.'"'
+    ]);
+  }
+
+  /**
+  * The function will check if the passed id is valid using:
+  * 1. Check for the null values
+  * 2. Check for the non numeric values
+  * 3. Check for the table id
+  **/
+  public function isValidTable($curTable){
+    if(is_null($curTable) || !is_numeric($curTable) || !$this->isCurTableExists($curTable)){
+      return false;
+    }
+    return true;
+  }
+
+  // Check weather the table with id=curTable exists
+  public function isCurTableExists($curTable){
+    try{
+      // returns exception if doesn't exists
+      $curTable = Table::findorFail($curTable);
```

Comment:
There is no exception being thrown in this, if the exception is being thrown in the findorFail method you don't need the try, if it is not and is only returning a string then you need to do something else.  

``` 
try { 
    if($curTable = Table::findorFail($curTable)){ 
        throw new Exception($curTable); 
    }
} catch(Exception $e) { 
  // May want to throw this to an error log as a warning or something or
  // the error exception message is completely pointless 
  // if doing something like this we need to make sure we are using it in a way to trace the bug or problem
  // not just generating a random exception that doesn't lead to something fruitful 
  // otherwise it might be good to just send that feedback to the user 
  echo "${$e->getMessage()}"; 
  return false 
}
```

Reply:
refactored